src/contexts/WorkDataContext.js

```diff
@@ -0,0 +1,60 @@
+import { createContext, useState } from 'react';
+import { DateTime } from 'luxon';
+
+export const WorkDataContext = createContext();
+
+const createFormDate = () => {
+  return new DateTime.utc();
+};
+
+const workDataArray = Array(1)
+  .fill(createFormDate())
+  .map((item) => ({ formDate: item.toISODate() }));
+
+const workDataMap = new Map();
+
+workDataArray.map(
+  ({ formDate, isOutside, isWelding, isScaffolding, workDetails }) => {
+    workDataMap.set(parseInt(formDate.replace(/-/g, '')), {
+      formDate,
+      isOutside,
+      isWelding,
+      isScaffolding,
+      workDetails,
+    });
+  }
+);
+
+const WorkDataContextProvider = ({ children }) => {
+  const [workValues, setWorkValues] = useState(
+    Array.from(workDataMap.values())
+  );
+
+  const submitWorkValues = (
```

Comment:
You're trying to modify workDataMap as an instance variable.  It should be a state.